### ISSUE TITLE
Refactoring for additional feature handling

### ISSUE THREAD

**Issue by johnharris85** [alexellis/derek#39]
Awesome project! Was thinking about taking a crack at #17 and wandered how you wanted to handle more features being added in the code? Checks like [this one](https://github.com/alexellis/derek/blob/master/main.go#L71) assume only one feature for handling PRs, and this might need to be refactored if more are being added?

Also the code in [handlePullRequest](https://github.com/alexellis/derek/blob/master/pullRequestHandler.go#L16) may need some refactoring to run through different features if they're enabled / disabled etc...

#17 is just a single addition so can fit into the current model, but just wanted to check on direction before I do anything, or if you think refactoring for potentially more scope later on is the way to go first? 

Happy to have a go at either, thanks!

**Comment by alexellis** [comment]
Hi @johnharris85,

Good to hear from you again - when you have time head over the jaas project again and checkout the latest features we've been adding.

Full disclosure - I'm unlikely to take refactoring changes before they are necessary for a feature / approved request. 

The highest priority tickets are marked with a label: https://github.com/alexellis/derek/labels/priority%2Fhigh

#17 is marked as low priority, but needs agreement on how it should work and be configured before any work is done on it. I've added some more context there for review

**Comment by johnharris85** [comment]
Thanks @alexellis! Ye wanted to ask here about refactoring before doing anything for that very reason :) Right now it looks like the codebase assumes only one 'feature' for PRs, so that function may turn into a long list of `if` statements if we add others (like #17). If those aren't high-priority / accepting PRs against right now then the refactoring likely isn't necessary yet. 

What's your view on requiring refactoring to support new 'features' (in addition to dco checking, as defined in the config yaml)?

**Comment by alexellis** [comment]
Simplest approach first, duplicate some statements when necessary, then review and if it makes sense after that introduce new abstractions.

**Comment by alexellis** [comment]
Since this is not an issue I'm going to close it but you can keep commenting.

**Comment by alexellis** [comment]
John would you like to work on this? https://github.com/alexellis/derek/issues/42 also I can't see you on our Slack community.. would you join to coordinate work? email alex@openfaas.com
